src/plan/generational/copying/global.rs

```diff
@@ -79,10 +78,11 @@ impl<VM: VMBinding> Plan for GenCopy<VM> {
         self.gen.last_collection_full_heap()
     }
 
-    fn gc_init(&mut self, heap_size: usize, vm_map: &'static VMMap) {
-        self.gen.gc_init(heap_size, vm_map);
-        self.copyspace0.init(vm_map);
-        self.copyspace1.init(vm_map);
+    fn get_spaces(&self) -> Vec<&dyn Space<Self::VM>> {
+        let mut ret = self.gen.get_spaces();
+        ret.push(&self.copyspace0);
+        ret.push(&self.copyspace1);
+        ret
```

Comment:
That sounds good. I don't think we need procedural macros. We could have a simple macro like
```rust
generate_get_spaces!(self.copyspace0, self.copyspace1; fallback = self.gen)
```

Reply:
No. That’s not good.  We use deriving macro so that we don’t repeat ourselves.  That means we declare the fields once (in the struct), and the macro generates related methods.  In your `generate_get_spaces!(self.copyspace0, self.copyspace1; fallback = self.gen)`, you repeated what the GenCopy struct already has, that is, copyspace0, copyspace1 and the parent `gen`.  I don’t think it is better or more concise than calling `gen.get_spaces` and add copyspaces{0,1} to the Vec.

Reply:
Considering that we may remove the `PlanTraceObject` and the macros/attributes for it, I don't think it is a good idea to adding another procedural macro for this one single use. Maybe we just keep it as it is now.

Reply:
OK. I am fine with the current code.